response_operations_ui/views/messages.py

```diff
@@ -121,14 +121,46 @@ def view_messages():
         return render_template("messages.html", breadcrumbs=breadcrumbs, response_error=True)
 
 
+@messages_bp.route('/threads/<thread_id>', methods=['GET'])
+@login_required
+def view_conversation(thread_id):
+    try:
+        thread_conversation = message_controllers.get_conversation(thread_id)['messages']
+        refined_thread = [_refine(message) for message in reversed(thread_conversation)]
+    except KeyError as ex:
+        logger.exception("A key error occurred")
+        raise ApiError(ex)
+    breadcrumbs = [
+        {"title": "Messages", "link": "/messages"},
+        {"title": refined_thread[0]["subject"]}
```

Comment:
Put this in a try/catch block.  If there were no threads returned, then you'll get an IndexError by trying to get the subject key on an empty list.  Maybe there is a better way then that.... maybe test the length of the refined thread and only make the subject breadcrumb if it's populated?  Might be worth asking the other guys what to do if that happens

Reply:
We could default it to 'Message' or something similar if subject is unavailable?

Reply:
Yea, that sounds good